### PR TITLE
Overhaul error handling / use gov uk templates in error pages / show auth error messages from interventions service

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

a few things:

moves around some template files to make things neater
removes the custom 'sanitizederror' class to favour 'HttpError' from 'http-errors' (which was already in use in the application)
handles different error codes in the error handler and shows the appropriate errorpage 
includes downstream api responses in non-prod error screens
includes accessError messages from interventions service in auth error screen
adds support for downstream APIs to send a 'userMessage' field in an error response
removes unused custom error handling code 

404 page:

Screenshot 2021-06-12 at 19 09 46

unauthorized page:

Screenshot 2021-06-12 at 22 46 09

unhandled error page in prod:

Screenshot 2021-06-12 at 19 42 27

unhandled error page in non-prod:

Screenshot 2021-06-12 at 21 56 02

## What is the intent behind these changes?

make errors nicer on the eyes for users.

### CODE REVIEW

File: server/utils/interventionsFormError.ts [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-// This re-throws the error if `interventionsServiceError` is not a validation error.
-export default function createFormValidationErrorOrRethrow(
-  interventionsServiceError: InterventionsServiceError
-): FormValidationError {
-  if (interventionsServiceError.validationErrors === undefined) {
-    throw interventionsServiceError
+// This re-throws the error if `err` is not an interventions service validation error.
+export default function createFormValidationErrorOrRethrow(err: Error): FormValidationError {
+  if (createError.isHttpError(err)) {
+    if (err.response?.body?.validationErrors) {
```
Comment (Gweaton) [inline server/utils/interventionsFormError.ts:11]: is it worth wrapping this in a private method/variable e.g. `isInterventionsServiceValidationError` to make it a bit more obvious what this condition is? `err.response?.body?.validationErrors` doesn't quite make it clear enough I think.

File: server/errorHandler.ts [hunk 2]
```diff
@@ -1,30 +1,54 @@
...
+      if (err.status === 403) {
+        res.status(403)
+
+        const args: Record<string, unknown> = { message: err.message }
+
+        // 403 responses from the interventions service contain further information in the
+        // response; if it's present, the authError template surfaces this to the end user
+        if (err.response) {
+          args.message = err.response.body?.message || args.message
+          args.accessErrors = err.response.body?.accessErrors
```
Comment (Gweaton) [inline server/errorHandler.ts:23]: As you showed in the demo, do we need to capitalize the first letter of these etc to make it a bit neater to users? Are you running these past the content team?

**Review by Gweaton — APPROVED** [review]
Looks good, a couple of small things, also would be great to get some tests in ASAP

File: server/errorHandler.ts (hunk 2 above)
Comment (tomsmyers) [reply server/errorHandler.ts:23]: yeh i'm going to do that once these changes are merged. then the next set of changes will just be cosmetic

File: server/utils/interventionsFormError.ts (hunk 1 above)
Comment (tomsmyers) [reply server/utils/interventionsFormError.ts:11]: yeh that's a good idea.

**Pull request merged**